### PR TITLE
NetworkInfo - Add Machine Indirection

### PR DESCRIPTION
**Pull request by manadart**
This is the beginning of some work to make the machine-based `NetworkInfo` less brittle.

We need to allow address sorting before materialising results into wire-types, losing the domain-type capability.

This simply adds an indirection for the unit's machine and populates it separately from address acquisition.

## QA steps

Non-functional changes; unit tests pass.

### CODE REVIEW

**Comment by manadart** [comment]
[NetworkInfo machine indirection](https://trello.com/c/VwRRcHpM/3718-networkinfo-machine-indirection)

File: apiserver/facades/agent/uniter/networkinfo.go [hunk 1]
```diff
@@ -60,7 +60,7 @@ func NewNetworkInfo(st *state.State, tag names.UnitTag) (NetworkInfo, error) {
 	return n, errors.Trace(err)
 }
 
-// NewNetworkInfoWithBehaviour initialises and returns a new NetworkInfo
+// NewNetworkInfoForStrategy initialises and returns a new NetworkInfo
```
Comment (hmlanigan) [inline apiserver/facades/agent/uniter/networkinfo.go:63]: While you're here, the comment block has no explanation of the retryFactory arg.

File: apiserver/facades/agent/uniter/networkinfoiaas.go [hunk 2]
```diff
@@ -9,18 +9,49 @@ import (
...
+
 // NetworkInfoIAAS is used to provide network info for IAAS units.
 type NetworkInfoIAAS struct {
 	*NetworkInfoBase
 
-	// machineNetworkInfos container network info for the unit's machine,
+	// machine is where the unit resides.
+	machine Machine
+
+	// machineNetworkInfos contains network info for the unit's machine,
```
Comment (hmlanigan) [inline apiserver/facades/agent/uniter/networkinfoiaas.go:54]: There is a subtle difference between this explanation and the method comment header block for populateMachineNetworkInfos.  The method indicates that only the unit's bound spaces will be included, not clear here.

**Review by hmlanigan — APPROVED** [review]
A few suggestions on the comments, LGTM.

File: apiserver/facades/agent/uniter/networkinfo.go (hunk 1 above)
Comment (manadart) [reply apiserver/facades/agent/uniter/networkinfo.go:63]: It is assigned directly to a member which has I think, a cogent comment.

File: apiserver/facades/agent/uniter/networkinfoiaas.go (hunk 2 above)
Comment (manadart) [reply apiserver/facades/agent/uniter/networkinfoiaas.go:54]: Good catch.

**Comment by manadart** [comment]
$$merge$$

**Pull request merged**